Mathlib/Analysis/Normed/Group/Quotient.lean

```diff
@@ -93,59 +93,206 @@ the previous paragraph kicks in.
+  uniformity_dist := by
+    rw [uniformity_eq_comap_nhds_one', (nhds_one_hasBasis.comap _).eq_biInf]
+    simp only [dist, preimage_setOf_eq, norm_eq_groupSeminorm, map_div_rev]
+
+variable (S) in
+/-- The quotient in the category of normed groups. -/
+@[to_additive "The quotient in the category of normed groups."]
+noncomputable instance instNormedCommGroup [hS : IsClosed (S : Set M)] :
+    NormedCommGroup (M ⧸ S) where
+  __ := MetricSpace.ofT0PseudoMetricSpace _
+
+-- This is a sanity check left here on purpose to ensure that potential refactors won't destroy
+-- this important property.
+example :
+    (instTopologicalSpaceQuotient : TopologicalSpace <| M ⧸ S) =
+      (instSeminormedCommGroup S).toUniformSpace.toTopologicalSpace := rfl
+
+example [IsClosed (S : Set M)] :
+   (instSeminormedCommGroup S) = NormedCommGroup.toSeminormedCommGroup := rfl
+
+end QuotientGroup
+
 open QuotientAddGroup Metric Set Topology NNReal
 
 variable {M N : Type*} [SeminormedAddCommGroup M] [SeminormedAddCommGroup N]
 
 /-- The definition of the norm on the quotient by an additive subgroup. -/
-noncomputable instance normOnQuotient (S : AddSubgroup M) : Norm (M ⧸ S) where
+@[deprecated QuotientAddGroup.instNorm (since := "2025-02-02")]
+noncomputable def normOnQuotient (S : AddSubgroup M) : Norm (M ⧸ S) where
   norm x := sInf (norm '' { m | mk' S m = x })
 
+@[deprecated QuotientAddGroup.norm_eq_infDist (since := "2025-02-02")]
 theorem AddSubgroup.quotient_norm_eq {S : AddSubgroup M} (x : M ⧸ S) :
-    ‖x‖ = sInf (norm '' { m : M | (m : M ⧸ S) = x }) :=
-  rfl
-
-theorem QuotientAddGroup.norm_eq_infDist {S : AddSubgroup M} (x : M ⧸ S) :
-    ‖x‖ = infDist 0 { m : M | (m : M ⧸ S) = x } := by
-  simp only [AddSubgroup.quotient_norm_eq, infDist_eq_iInf, sInf_image', dist_zero_left]
-
-/-- An alternative definition of the norm on the quotient group: the norm of `((x : M) : M ⧸ S)` is
-equal to the distance from `x` to `S`. -/
-theorem QuotientAddGroup.norm_mk {S : AddSubgroup M} (x : M) :
-    ‖(x : M ⧸ S)‖ = infDist x S := by
-  rw [norm_eq_infDist, ← infDist_image (IsometryEquiv.subLeft x).isometry,
-    IsometryEquiv.subLeft_apply, sub_zero, ← IsometryEquiv.preimage_symm]
-  congr 1 with y
-  simp only [mem_preimage, IsometryEquiv.subLeft_symm_apply, mem_setOf_eq, QuotientAddGroup.eq,
-    neg_add, neg_neg, neg_add_cancel_right, SetLike.mem_coe]
+    ‖x‖ = sInf (norm '' { m : M | (m : M ⧸ S) = x }) := by
+  simp only [norm_eq_infDist, infDist_eq_iInf, sInf_image', dist_zero_left]
 
+@[deprecated "Replaced by a private lemma" (since := "2025-02-02")]
 theorem image_norm_nonempty {S : AddSubgroup M} (x : M ⧸ S) :
     (norm '' { m | mk' S m = x }).Nonempty :=
   .image _ <| Quot.exists_rep x
 
+@[deprecated norm_nonneg (since := "2025-02-02")]
 theorem bddBelow_image_norm (s : Set M) : BddBelow (norm '' s) :=
   ⟨0, forall_mem_image.2 fun _ _ ↦ norm_nonneg _⟩
 
+@[deprecated "No replacement" (since := "2025-02-02")]
```

Comment:
Why is "no replacement" ok? Should this lemma exist for `infDist`?

Reply:
1. This doesn't seem like a particularly useful lemma
2. This lemma already exists for `⨅`, up to trivial simplifications

Reply:
I think the comment should suggest the rough replacement, even if there's no direct lemma.

Reply:
I've written "See constituent lemmas". Isn't that enough?

Reply:
I'm not really sure what that really means here, and similar for the "Replaced by a private lemma" earlier. The idea of deprecation messages is to make it straight-forward for end-users of mathlib to amend their code to the new changes. 

This proof has also gone from a one-line proof to a `simp only / exact / simpa / aesop`, so it's not particularly nice for someone to see which part of the proof is the correct replacement.

It looks to me like there should at least be a lemma saying that `infDist` is the glb of some set, and then you can deprecate in favour of that.